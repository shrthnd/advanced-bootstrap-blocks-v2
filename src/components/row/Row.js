export const Row = (props) => {
  const { className, children } = props
  return <div className={['row', className].join(' ').trim()}>{children}</div>
}

export default Row
