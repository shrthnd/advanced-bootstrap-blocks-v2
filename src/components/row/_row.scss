[data-type="advanced-bootstrap-blocks/row"] {
  margin-top: 0;
  margin-bottom: 0;
  max-width: 100%;
  
  > .row { 
    outline: 1px dashed orange;
  
    > .block-editor-inner-blocks {
      width: 100%;
    }
  
    > .block-editor-inner-blocks > .block-editor-block-list__layout {
      display: flex;
      width: 100%;
      min-height: 30px;
    }
  
    > .block-editor-inner-blocks > .block-editor-block-list__layout > .wp-block {
      // [data-type="advanced-bootstrap-blocks/column"] or other child blocks... 
    } 
  }
}
