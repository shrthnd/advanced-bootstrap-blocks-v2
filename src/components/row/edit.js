import Row from './Row'

export const edit = ({ className }) => (
  <Row className={className}>
    <InnerBlocks />
  </Row>
)

export default edit
