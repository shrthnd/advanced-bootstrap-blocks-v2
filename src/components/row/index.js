import { registerBlockType, getBlockDefaultClassName } from '@wordpress/blocks'
import { InnerBlocks } from '@wordpress/block-editor'

import { defaultBlockSettings } from '../../utils'
import Row from './Row'

const BOOTSTRAP_BLOCK_TYPE = 'advanced-bootstrap-blocks/row'
const defaultClassName = getBlockDefaultClassName(BOOTSTRAP_BLOCK_TYPE)

/*
 * Register Bootstrap 4 - Row block
 */
registerBlockType(BOOTSTRAP_BLOCK_TYPE, {
  ...defaultBlockSettings({
    title: 'BS4 - Row',
  }),
  edit: ({ className }) => (
    <Row className={className}>
      <InnerBlocks />
    </Row>
  ),
  save: ({ className }) => (
    <Row className={className}>
      <InnerBlocks.Content />
    </Row>
  ),
})

/*
 * Remove default block className from block className list
 */
const setBlockCustomClassName = (blockName) => {
  return blockName === defaultClassName ? [] : blockName
}
wp.hooks.addFilter(
  'blocks.getBlockDefaultClassName',
  `${BOOTSTRAP_BLOCK_TYPE}/set-block-custom-class-name`,
  setBlockCustomClassName
)
