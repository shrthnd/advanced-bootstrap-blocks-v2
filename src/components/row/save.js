import Row from './Row'

export const save = ({ className }) => (
  <Row className={className}>
    <InnerBlocks.Content />
  </Row>
)

export default save
