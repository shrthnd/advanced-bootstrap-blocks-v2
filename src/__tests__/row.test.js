import React from 'react'
import { shallow } from 'enzyme'
import { Row } from '../components/row/Row'

describe('Row', () => {
  it('should accept custom className', function () {
    const comparison = shallow(<Row className="align-items-center" />).contains(
      <div className="row align-items-center"></div>
    )
    expect(comparison).toBe(true)
  })

  it('should match snapshot', function () {
    expect(
      shallow(<Row className="align-items-center">Test</Row>)
    ).toMatchSnapshot()
  })
})
