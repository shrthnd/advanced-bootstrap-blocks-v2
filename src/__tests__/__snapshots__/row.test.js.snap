// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row should match snapshot 1`] = `
<div
  className="row align-items-center"
>
  Test
</div>
`;
